### PR TITLE
import private key: importPrivateKeypage, button on createWallet page…

### PR DESCRIPTION
**Pull request by soshochang**
import private key: 
-accountController background script - add code to recover wallet from prKey and log user in
-re-factor importMnemonic and addAccountAndLogin code
-prKey input validation (empty, existing wallet, and secp validation) and error dialog
-combine importMnemonic page w/ importPrKey page, rename to importWallet

### CODE REVIEW

File: src/background/controllers/accountController.ts [hunk 1]
```diff
@@ -286,12 +306,11 @@ export default class AccountController extends IController {
   }
 
   /*
-  * Checks if a wallet mnemonic has already been taken.
-  * @param mnemonic The wallet mnemonic to check.
-  * @return Has the mnemonic been used.
+  * Checks if a wallet is already in the mainnet or testnet accounts list.
+  * @param privateKeyHash, unique and deterministic for each wallet.
```
Comment (dwalintukan) [inline src/background/controllers/accountController.ts]: remove `,`

File: src/popup/components/BorderTextField/styles.ts [hunk 2]
```diff
@@ -13,8 +13,7 @@ const styles: StyleRulesCallback = (theme: Theme) => ({
     borderRadius: theme.border.radius,
   },
   textFieldInput: {
-    fontSize: theme.font.sm,
-    fontWeight: theme.fontWeight.bold,
```
[the remark this answers is not in the document]
Comment (soshochang) [reply src/popup/components/BorderTextField/styles.ts]: fixed

File: src/popup/pages/ImportMnemonic/styles.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ const styles: StyleRulesCallback = (theme: Theme) => ({
   },
   mnemonicFieldInput: {
     padding: theme.padding.sm,
-    fontSize: theme.font.lg,
+    fontSize: theme.font.md,
```
[the remark this answers is not in the document]
Comment (soshochang) [reply src/popup/pages/ImportMnemonic/styles.ts]: lili and I discussed changing the design a little bit

File: src/popup/pages/ImportPrivateKey/index.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        <NavBar hasNetworkSelector title="" />
+        <div className={classes.contentContainer}>
+          <Typography className={classes.headerText}>Import Private Key</Typography>
+          <div className={classes.inputContainer}>
+            <div className={classes.fieldContainer}>
+              <TextField
+                className={classes.privateKeyTextField}
+                autoFocus
+                required
+                multiline
```
[the remark this answers is not in the document]
Comment (soshochang) [reply src/popup/pages/ImportPrivateKey/index.tsx]: it takes 2 lines to fit on the text input field though?

File: src/popup/pages/ImportPrivateKey/index.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+const ErrorDialog: React.SFC<any> = observer(({ store: { importStore }}: any) => (
+  <Dialog
+    disableBackdropClick
+    open={importStore.importPrivateKeyFailed}
+    onClose={() => importStore.importPrivateKeyFailed = false}
+  >
+    <DialogTitle>Invalid Private Key</DialogTitle>
+    <DialogContent>
+      <DialogContentText>This wallet has already been imported.</DialogContentText>
```
[the remark this answers is not in the document]
Comment (soshochang) [reply src/popup/pages/ImportPrivateKey/index.tsx]: well technically we don't know if it was imported as a private key or a mnemonic previously, so I chose the more generic language

File: src/popup/pages/ImportPrivateKey/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (dwalintukan) [reply src/popup/pages/ImportPrivateKey/index.tsx]: it doesn't wrap the word? technically, if the word will break and wrap it is still a single line (I believe?).

File: src/popup/pages/ImportPrivateKey/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (dwalintukan) [reply src/popup/pages/ImportPrivateKey/index.tsx]: gotcha, makes sense.

File: src/constants.ts [hunk 6]
```diff
@@ -28,10 +28,10 @@ export enum MESSAGE_TYPE {
   LOGIN_SUCCESS_WITH_ACCOUNTS,
   LOGIN_SUCCESS_NO_ACCOUNTS,
   LOGIN_FAILURE,
-  CREATE_WALLET,
   SAVE_TO_FILE,
   IMPORT_MNEMONIC,
-  IMPORT_MNEMONIC_FAILURE,
+  IMPORT_PRIVATE_KEY,
+  IMPORT_MNEMONIC_PRKEY_FAILURE,
```
[the remark this answers is not in the document]
Comment (soshochang) [reply src/constants.ts]: hmm it is shorter, but I kind of think it becomes less clear that it is related to  IMPORT_MNEMONIC_FAILURE, and IMPORT_PRIVATE_KEY

[the remark this answers is not in the document]
Comment (dwalintukan) [reply src/constants.ts]: FYI, when i comment with `nit:` i am basically saying its up to you if you want to change.

**Comment by soshochang** [comment]
okay

On Tue, Aug 14, 2018 at 10:36 PM, Deric Walintukan <notifications@github.com

**Pull request merged**